packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeNodeRenderer.tsx

```diff
@@ -112,23 +112,22 @@ const HierarchyNode = memo(
     const { localizedStrings } = useLocalizationContext();
 
     const nodeActions = useMemo(() => {
-      if (!actions || actions.length === 0) {
-        return undefined;
-      }
-
       const actionButtons: ReactElement[] = [];
-
-      if (error && error.error.type === "Unknown") {
+      if (error && error.error.type === "Unknown" && reloadTree) {
         actionButtons.push(
           <TreeActionButton
             label={localizedStrings.retry}
-            action={() => reloadTree?.({ parentNodeId: node.id, state: "reset" })}
+            action={() => reloadTree({ parentNodeId: node.id, state: "reset" })}
             show={true}
             icon={refreshSvg}
           />,
         );
       }
 
+      if (!actions || actions.length === 0) {
+        return actionButtons;
+      }
```

Comment:
Do you think this is necessary? The below will be a no-op anyway if `actions` is empty.

Reply:
I think we still need to check that actions are not undefined, because you can't call map on undefined even with `?`.